lib/control.ml

```diff
@@ -16,6 +16,8 @@ let steps = ref 0
 
 let enable_thread_delay = ref false
 
+exception Signal of int
```

Comment:
Why the int argument? You always raise it with the same value?

Reply:
Yes, this is an attempt at generalization. Maybe some day someone wants to use a signal with a different code. Then I want to make sure that the `timeout` function does not catch it (by checking for the correct value).

Reply:
Well, instead of introducing dead code I think you should define a dedicated exception for SIGALRM signals here. Some day seems too far away...

Reply:
Done.